File: LeetcodePractices/Simple/MergeTwoSortedList/C/mergetwosortedlist.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
#include "stdio.h"
 struct ListNode {
      int val;
      struct ListNode *next;
 };
 
struct ListNode* mergeTwoLists(struct ListNode* l1, struct ListNode* l2){
    if(l1 !=NULL && l2 !=NULL)
    {
        if(l1->val < l2->val)                                               //pointer:"->", class:"."
        {
            l1->next =  mergeTwoLists(l1->next,l2);
            return l1;
        }
        else
        {
            l2->next = mergeTwoLists(l2->next, l1);
            return l2;
        }
    }

    if(l1 != NULL)
    {
        return l1;
    }

    if(l2 != NULL)
    {
        return l2;
    }

    return NULL;                                                            //need return for non-void function

}

File: LeetcodePractices/Simple/MergeTwoSortedList/Python/mergetwosortedlist.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def mergeTwoLists(self, l1, l2):
        """
        :type l1: ListNode
        :type l2: ListNode
        :rtype: ListNode
        """
        if (l1 != None and l2 != None):
            if (l1.val < l2.val):
                l1.next = self.mergeTwoLists(l1.next, l2)           #python could not auto extend local scope, so need "self." to recognize the func in class definition
                return l1
            else:
                l2.next = self.mergeTwoLists(l2.next, l1)
                return l2
        if (l1 != None) :
            return l1
        if (l2 != None) :
            return l2